### DISCUSSION TITLE
How would I run a system at a fixed time interval that can be changed at runtime

### DISCUSSION THREAD

**Discussion by rayanmargham** [bevyengine/bevy#8382] [Q&A]
Heyo Everyone! so I'm working on a bevy video plugin using ffmpeg-next! I've got support working for 2D materials and I'm currently working on 3D material support. I need to constantly reassign the base_color_texture variable every video frame and therefore I need a system that can run every video (not game) frame. I'm just wondering if theres a way to run a system every insert fixed time interval here and be able to change that time interval on runtime. I'm using bevy 0.10

I'll publish the plugin as a crate on crates.io once I'm done :D

**Comment by mnett82** [comment]
I'm not sure if there's a built-in solution that serves your need. There's [on_timer](https://docs.rs/bevy/latest/bevy/time/common_conditions/fn.on_timer.html), but this accepts only a static interval.

You could throttle your system by keeping a cooldown timer via `Local<Duration>`, but IIUC, your main concern is that you want to run the system more frequently than the render loop?

Why would you need to reassign colors more often than your scene is rendered though? The intermittent changes would never be visible to a user. Or are you using color changes to do something different than driving visuals?
